src/vim-loader/document.ts

```diff
@@ -217,6 +257,144 @@ export class Document {
     return this._strings[index]
   }
 
+  /**
+   * Returns all parameters of an element and of its family type and family
+   * @param element element index
+   * @returns An array of paramters with name, value, group
+   */
+  async getElementParameters (element: number): Promise<
+    {
+      name: string
+      value: string
+      group: string
+    }[]
+  > {
+    const elements = new Set<number>()
+    elements.add(element)
+
+    const familyInstance = await this.getElementFamilyInstance(element)
+    if (familyInstance !== undefined) {
+      const familyType = await this.getFamilyInstanceFamilyType(familyInstance)
+      const family = await this.getFamilyTypeFamily(familyType)
+
+      const familyTypeElement = await this.getFamiltyTypeElement(familyType)
+      elements.add(familyTypeElement)
+
+      const familyElement = await this.getFamilyElement(family)
+      elements.add(familyElement)
+    }
+
+    const result = await this.getElementsParameters(elements)
+    return result
+  }
+
+  private async getElementsParameters (elements: Set<number>) {
+    const parameterTable = await this._entities.getBfast(
+      objectModel.parameter.table
+    )
+    const parameterElement = await parameterTable.getArray(
+      objectModel.element.index
+    )
+    const parameterValue = await parameterTable.getArray(
+      objectModel.parameter.columns.value
+    )
+    const parameterDescription = await parameterTable.getArray(
+      objectModel.parameterDescriptor.index
+    )
+
+    const parameterDescriptor = await this._entities.getBfast(
+      objectModel.parameterDescriptor.table
+    )
+    const parameterDescriptorName = await parameterDescriptor.getArray(
+      objectModel.parameterDescriptor.columns.name
+    )
+    const parameterDescriptorGroup = await parameterDescriptor.getArray(
+      objectModel.parameterDescriptor.columns.group
+    )
+
+    const result: {
+      name: string
+      value: string
+      group: string
+    }[] = []
+
+    parameterElement.forEach((e, i) => {
+      if (elements.has(e)) {
+        const value = this._strings[parameterValue[i]].split('|')
+        const displayValue = value[value.length - 1]
```

Comment:
Suggestion: fall back to the native value (i.e. value[0]) if the display value does not exist